contracts/mainnet/connectors/sushi-incentive/main.sol

```diff
@@ -70,12 +74,17 @@ abstract contract SushipswapIncentiveResolver is Helpers, Events {
         token1 = changeEthAddrToWethAddr(token1);
         token2 = changeEthAddrToWethAddr(token2);
         amount = getUint(getId, amount);
-        (uint256 _pid, uint256 _version, ) = _getPoolId(token1, token2);
-        setUint(setId, _pid);
-        require(_pid != uint256(-1), "pool-does-not-exist");
-        _withdraw(_pid, amount, _version);
+        if(data.poolId == uint256(-1)|| data.version <= 0){
+            data = _getPoolId(
+                token1,
+                token2
+            );
+        }
+        setUint(setId, data.poolId);
```

Comment:
@pradyuman-verma I think, we need to use set amount withdrawn instead of `poolId`